crates/sui-framework/packages/sui-framework/sources/balance.move

```diff
@@ -47,6 +47,12 @@ module sui::balance {
         Supply { value: 0 }
     }
 
+    /// Destroy an empty supply
+    public fun destroy_zero_supply<T: drop>(self: Supply<T>) {
```

Comment:
Please, add tests

Reply:
I added the `T: drop` because `create_supply` requires it. Adding tests

Reply:
The `T: drop` is there because `create_supply` has it.

Reply:
Added tests. @damirka do you think it is sound to remove the `drop`?